src/app/core/filter/list-filter/list-filter.component.ts

```diff
@@ -1,33 +1,28 @@
 import { Component, EventEmitter, Input, Output } from "@angular/core";
-import { Filter, SelectableFilter } from "../filters/filters";
 import { Entity } from "../../entity/model/entity";
 import { MatFormFieldModule } from "@angular/material/form-field";
 import { MatSelectModule } from "@angular/material/select";
 import { BorderHighlightDirective } from "../../common-components/border-highlight/border-highlight.directive";
-import { NgForOf } from "@angular/common";
+import { JsonPipe, NgForOf } from "@angular/common";
+import { ReactiveFormsModule } from "@angular/forms";
+import { SelectableFilter } from "../filters/filters";
 
 @Component({
   selector: "app-list-filter",
   templateUrl: "./list-filter.component.html",
   imports: [
     MatFormFieldModule,
     MatSelectModule,
+    ReactiveFormsModule,
     BorderHighlightDirective,
     NgForOf,
+    JsonPipe,
   ],
   standalone: true,
 })
 export class ListFilterComponent<E extends Entity> {
-  @Input()
-  public set filterConfig(value: Filter<E>) {
-    this._filterConfig = value as SelectableFilter<E>;
-  }
-  _filterConfig: SelectableFilter<E>;
-  @Input() selectedOption: string;
-  @Output() selectedOptionChange = new EventEmitter<string>();
-
-  selectOption(selectedOptionKey: string) {
-    this.selectedOption = selectedOptionKey;
-    this.selectedOptionChange.emit(selectedOptionKey);
-  }
+  @Input({ transform: (value: any) => value as SelectableFilter<E> })
```

Comment:
What we need to be a bit careful with. Currently we also use the `@Input` annotations in the dynamic components but the `transform` function is not (yet) used. 
In this case it is fine because the component is not dynamically used and also the transform function doesn't really do anything but just as a reminder.

Reply:
todo: create analyse issue here

Reply:
https://github.com/Aam-Digital/ndb-core/issues/2176